dbt/adapters/snowflake/relation.py

```diff
@@ -33,3 +51,48 @@ def is_dynamic_table(self) -> bool:
     @classproperty
     def DynamicTable(cls) -> str:
         return str(SnowflakeRelationType.DynamicTable)
+
+    @classproperty
+    def get_relation_type(cls) -> Type[SnowflakeRelationType]:
+        return SnowflakeRelationType
+
+    @classmethod
+    def from_runtime_config(cls, runtime_config: RuntimeConfigObject) -> RelationConfigBase:
+        model_node: ModelNode = runtime_config.model
+        relation_type = SnowflakeRelationType(model_node.config.materialized)
+
+        if relation_config := cls.relation_configs.get(relation_type):
```

Comment:
Oh this operator. Oh good.

What's your intended semantics? Non-`Optional` assignment semantics frighten me.

Reply:
In this case we control the value, and it can't be `None` because relation_type is a `Enum`. But I do explicitly use that in all of the config parsers. Passing `None` is effectively not passing it at all. I think I covered the scenarios there, where `None` is not actually a valid value for that config parameter. But agreed, we need to watch out for that and check for the existence of the key in those scenarios. I could also make the argument that we don't allow that value in our interface.